### ISSUE TITLE
NetworkManager overwriting /etc/resolv.conf when dns set to "unbound"

### ISSUE THREAD

**Issue by wkral** [NixOS/nixpkgs#61645]
## Issue description
I have unbound set up to do DNS resolution which allows me to configure a local private domain  for development work. I have had this setup for quite some time and it worked previously, I think it may have changed with a recent networkmanager upgrade.

According to the NetworkManager doc:

> unbound: NetworkManager will talk to unbound and dnssec-triggerd, providing a "split DNS" configuration with DNSSEC support. /etc/resolv.conf will be managed by dnssec-trigger daemon.

However `/etc/resolv.conf` gets overwritten with a comment saying NetworkManager did it.

### Steps to reproduce

From `configuration.nix`:
```
networking.networkmanager.enable = true;
services.unbound.enable = true;
```

After a `nixos-rebuild switch` `/etc/resolv.conf` looks like:
```
# Generated by resolvconf
nameserver 127.0.0.1
options edns0
```

When I connect to a wireless network a few moments later `/etc/resolv.conf` file will look like:
```
# Generated by NetworkManager
nameserver 8.8.8.8
nameserver 8.8.4.4
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.
 - system: `"x86_64-linux"`
 - host os: `Linux 5.1.2, NixOS, 19.09pre179853.bc9df0f6611 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.09pre179853.bc9df0f6611"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by dtzWill** [comment]
Yeah, we changed a number of settings re:networkmanager during the move to meson and version bump-- this in particular is what I was worried about when putting together this: https://github.com/NixOS/nixpkgs/pull/60904 .  I think we usually set the related values explicitly in configs we generate which is why it has gone unnoticed (I hadn't wrangled it down to a solid repro yet, actually, because of this and time/energy :)).

Anyway it seems quite likely we changed the behavior here.  I don't know what behavior we "want" but if no one else has any ideas it sounds like "do what we did before" is the favored solution :).

Let's not sit on this too long, please! Thanks for reporting!

**Comment by angristan** [comment]
Is it related to https://github.com/NixOS/nixpkgs/issues/61490 ?

**Comment by wkral** [comment]
I saw that issue, but mine seems to be a little different as rebuild it doing the correct thing, and the setting in the NetworkManager.conf from the store is also correct, but it seems to be related to what @dtzWill showed as there are missing flags and dependencies from the meson build which means the installed NetworkManager can't actually properly comply with the "unbound" setting. Or at least that's my interpretation.

I think @dtzWill should fix it with the mentioned PR. Thanks for that by the way.

**Comment by turboMaCk** [comment]
I've run into this issue on my desktop currently making it impossible for me to rebuild the system configuration. Is there any known workaround to fix this?

**Comment by turboMaCk** [comment]
This comment saved me https://github.com/NixOS/nixpkgs/issues/61490#issuecomment-494832451

**Comment by wkral** [comment]
I think this is fixed by 80acb28 as well, tried reproducing after upgrading to `19.09pre184855.beff2f8d75e (Loris)` and my `resolv.conf` doesn't change after connecting to wifi network.

Closing.
